src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/SimpleSVType.java

```diff
@@ -50,101 +64,172 @@ private static String getIDString(final NovelAdjacencyAndAltHaplotype novelAdjac
 
+            final SimpleInterval dupSeqRepeatUnitRefSpan = ((BreakpointComplications.SmallDuplicationBreakpointComplications)
+                    novelAdjacencyAndAltHaplotype.getComplication()).getDupSeqRepeatUnitRefSpan();
+
             return GATKSVVCFConstants.DUP_TAN_EXPANSION_INTERNAL_ID_START_STRING + GATKSVVCFConstants.INTERVAL_VARIANT_ID_FIELD_SEPARATOR
-                    + novelAdjacencyAndAltHaplotype.getLeftJustifiedLeftRefLoc().getContig() + GATKSVVCFConstants.INTERVAL_VARIANT_ID_FIELD_SEPARATOR
-                    + novelAdjacencyAndAltHaplotype.getLeftJustifiedLeftRefLoc().getEnd() + GATKSVVCFConstants.INTERVAL_VARIANT_ID_FIELD_SEPARATOR
-                    + novelAdjacencyAndAltHaplotype.getLeftJustifiedRightRefLoc().getStart();
+                    + makeLocationString(dupSeqRepeatUnitRefSpan.getContig(), dupSeqRepeatUnitRefSpan.getStart(),
+                    dupSeqRepeatUnitRefSpan.getContig(), dupSeqRepeatUnitRefSpan.getEnd());
         }
     }
 
     public static final class ImpreciseDeletion extends SimpleSVType {
 
         @Override
         public String toString() {
-            return TYPES.DEL.name();
+            return SupportedType.DEL.name();
         }
 
-        public ImpreciseDeletion(final EvidenceTargetLink evidenceTargetLink, final int svLength, final ReadMetadata metadata) {
+        public ImpreciseDeletion(final EvidenceTargetLink evidenceTargetLink, final int svLength, final ReadMetadata metadata,
+                                 final ReferenceMultiSource reference) {
 
-            super(getIDString(evidenceTargetLink, metadata),
+            super(metadata.getContigName(evidenceTargetLink.getPairedStrandedIntervals().getLeft().getInterval().getContig()),
+                    evidenceTargetLink.getPairedStrandedIntervals().getLeft().getInterval().midpoint(),
+                    evidenceTargetLink.getPairedStrandedIntervals().getRight().getInterval().midpoint(),
+                    getIDString(evidenceTargetLink, metadata),
+                    Allele.create(getRefBases(evidenceTargetLink, metadata, reference), true),
                     Allele.create(createBracketedSymbAlleleString(GATKSVVCFConstants.SYMB_ALT_ALLELE_DEL)),
                     svLength,
-                    noExtraAttributes);
+                    Collections.singletonMap(GATKSVVCFConstants.IMPRECISE, true));
+        }
+
+        private static byte[] getRefBases(final EvidenceTargetLink evidenceTargetLink, final ReadMetadata metadata,
+                                          final ReferenceMultiSource reference) {
+            return extractRefBases(
+                    new SimpleInterval(metadata.getContigName(evidenceTargetLink.getPairedStrandedIntervals().getLeft().getInterval().getContig()),
+                            evidenceTargetLink.getPairedStrandedIntervals().getLeft().getInterval().midpoint(),
```

Comment:
I'd pull this variable out so you don't have to get it through the same chain of getters twice and it will have a descriptive name.

Reply:
done